llvm/include/llvm/Transforms/Utils/ScalarEvolutionExpander.h

```diff
@@ -418,6 +418,13 @@ class SCEVExpander : public SCEVVisitor<SCEVExpander, Value *> {
   BasicBlock::iterator findInsertPointAfter(Instruction *I,
                                             Instruction *MustDominate) const;
 
+  /// If \p L contains exits which may execute conditionally and contain UDiv
+  /// expressions with divisors that can be 0, expanding \p BTC may introduce
+  /// new UB. In that case, rewrite UDiv(A, B) to  UDiv(A, UMAX(1, B)). If B is
+  /// 0, that exit cannot be taken.
```

Comment:
Does this assume the loop eventually exits?  If it does, probably worth explicitly noting.